### PR TITLE
[WIP] add shallow delete route

### PR DESCRIPTION
**Pull request by ashleygwilliams**

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
```
Failed tests:

  237) Project Handlers DELETE - shallow fail, not moderator:

      Cannot read property 'credentials' of undefined

      at internals.Connection.inject (/Users/ag_dubs/Projects/api.webmaker.org/node_modules/hapi/lib/connection.js:256:17)
      at internals.inject (/Users/ag_dubs/Projects/api.webmaker.org/node_modules/hapi/lib/plugin.js:368:32)
      at /Users/ag_dubs/Projects/api.webmaker.org/test/services/api/handlers/projects.js:536:14
      at Immediate._onImmediate (/Users/ag_dubs/Projects/api.webmaker.org/node_modules/lab/lib/runner.js:453:17)
      at processImmediate [as _immediateCallback] (timers.js:367:17)
```

**Comment by ashleygwilliams** [comment]
also:
```
Coverage: 99.94% (2/3592)
services/api/handlers/projects.js missing coverage on line(s): 263, 264
Code coverage below threshold: 99.94 < 100
```

but i am using the `server.methods.projects.remove` and if i write another test using that i get yelled at. doesn't make sense to test again anyways IMO. thoughts @cadecairos ?

File: services/api/handlers/projects.js [hunk 1]
```diff
@@ -254,6 +254,27 @@ exports.patch = {
   }
 };
 
+exports.delShallow = function(request, reply) {
+  request.server.methods.projects.remove(
+    [
+      request.params.project
+    ],
+    function(err, result) {
+      if(err) {
```
Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: coveralls is upset about this line not being tested, but it *is*, here: https://github.com/mozilla/api.webmaker.org/blob/develop/test/services/api/handlers/projects.js#L1835-L1848

Comment (cadecairos) [inline services/api/handlers/projects.js]: That test wouldnt trigger this code though, unless I'm missing something

Comment (cadecairos) [inline services/api/handlers/projects.js]: you need to write a similar test to that which triggers the error condition in this callback I think.

Comment (ashleygwilliams) [inline services/api/handlers/projects.js]: to test this wouldn't i just stub `remove` again though? when i do it i get an error about "re-wrapping' `remove`, which frankly i didn't fully understand, heh.

**Comment by ashleygwilliams** [comment]
```
Failed tests:

  237) Project Handlers DELETE - shallow fail, not moderator:

      actual expected

      200403

      Expected 200 to equal specified value

      at /Users/ag_dubs/Projects/api.webmaker.org/test/services/api/handlers/projects.js:537:36
```

**Comment by ashleygwilliams** [comment]
so basically we have to think about auth now, but otherwise this should be complete.

File: services/api/handlers/projects.js (hunk 1 above)
Comment (cadecairos) [inline services/api/handlers/projects.js]: it's only stubbed for the duration of that test case

Comment (cadecairos) [inline services/api/handlers/projects.js]: the double wrapping error sounds like it is being stubbed somewhere and restore isn't song called on it. might be an existing bug

**Pull request closed without merging**